public/pages/QueryGroupDetails/QueryGroupDetails.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CoreStart } from 'opensearch-dashboards/public';
+// @ts-ignore
+import Plotly from 'plotly.js-dist';
+import hash from 'object-hash';
+import { useParams, useHistory, useLocation } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import {
+  EuiButton,
+  EuiCodeBlock,
+  EuiFlexGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiIconTip,
+} from '@elastic/eui';
+import { QUERY_INSIGHTS } from '../TopNQueries/TopNQueries';
+import { QueryGroupAggregateSummary } from './Components/QueryGroupAggregateSummary';
+import { QueryGroupSampleQuerySummary } from './Components/QueryGroupSampleQuerySummary';
+
+export const QueryGroupDetails = ({ queries, core }: { queries: any; core: CoreStart }) => {
+  const { hashedQuery } = useParams<{ hashedQuery: string }>();
+  const query = queries.find((q: any) => hash(q) === hashedQuery);
+
+  const convertTime = (unixTime: number) => {
+    const date = new Date(unixTime);
+    const loc = date.toDateString().split(' ');
+    return loc[1] + ' ' + loc[2] + ', ' + loc[3] + ' @ ' + date.toLocaleTimeString('en-US');
+  };
+
+  const history = useHistory();
+  const location = useLocation();
+
+  useEffect(() => {
+    core.chrome.setBreadcrumbs([
+      {
+        text: 'Query insights',
+        href: QUERY_INSIGHTS,
+        onClick: (e) => {
+          e.preventDefault();
+          history.push(QUERY_INSIGHTS);
+        },
+      },
+      { text: `Query group details: ${convertTime(query.timestamp)}` },
+    ]);
+  }, [core.chrome, history, location, query.timestamp]);
+
+  useEffect(() => {
+    let x: number[] = Object.values(query.phase_latency_map);
+    if (x.length < 3) {
+      x = [0, 0, 0];
+    }
+    const data = [
+      {
+        x: x.reverse(),
+        y: ['Fetch    ', 'Query    ', 'Expand    '],
+        type: 'bar',
+        orientation: 'h',
+        width: 0.5,
+        marker: { color: ['#F990C0', '#1BA9F5', '#7DE2D1'] },
+        base: [x[2] + x[1], x[2], 0],
+        text: x.map((value) => `${value}ms`),
+        textposition: 'outside',
+        cliponaxis: false,
+      },
+    ];
+    const layout = {
+      autosize: true,
+      margin: { l: 80, r: 80, t: 25, b: 15, pad: 0 },
+      autorange: true,
+      height: 120,
+      xaxis: {
+        side: 'top',
+        zeroline: false,
+        ticksuffix: 'ms',
+        autorangeoptions: { clipmin: 0 },
+        tickfont: { color: '#535966' },
+        linecolor: '#D4DAE5',
+        gridcolor: '#D4DAE5',
+      },
+      yaxis: { linecolor: '#D4DAE5' },
+    };
+    const config = { responsive: true };
+    Plotly.newPlot('latency', data, layout, config);
+  }, [query]);
```

Comment:
Not blocking, but can we move this out to a separate component? Also have we tested this with dark mode/other dashboard configurations and think about how not to hard code these values?